src/app/core/admin/admin-entity-details/admin-entity-field/admin-entity-field.component.ts

```diff
@@ -273,19 +307,34 @@ export class AdminEntityFieldComponent implements OnInit {
     this.createNewAdditionalOption = undefined;
   }
 
-  save() {
+  async save() {
     this.form.markAllAsTouched();
-    if (this.form.invalid) {
-      return;
-    }
+    if (this.form.invalid) return;
 
     const fieldId = this.fieldIdForm.getRawValue();
+    let shouldUpdateBaseSchema = false;
+
+    if (this.publicFormConfig) {
+      const publicFormConfigUpdated =
+        await this.publicFormService.updateFieldInPublicFormConfig(
+          this.publicFormConfig,
+          fieldId,
+          this.entitySchemaField,
+        );
+
+      await this.entityMapper.save(this.publicFormConfig, false);
+      shouldUpdateBaseSchema = !publicFormConfigUpdated;
+    } else {
+      shouldUpdateBaseSchema = true;
+    }
 
-    this.adminEntityService.updateSchemaField(
-      this.entityType,
-      fieldId,
-      this.entitySchemaField,
-    );
+    if (shouldUpdateBaseSchema) {
+      this.adminEntityService.updateSchemaField(
+        this.entityType,
+        fieldId,
+        this.entitySchemaField,
+      );
+    }
 
     this.dialogRef.close(fieldId);
```

Comment:
I am wondering if we can implement the "interface" of AdminEntityField to be that it just returns from the dialog with the entity-field schema, so that we don't need to do special logic inside here for public-forms or other cases :thinking: 

(then the component calling this dialog would need to take care of saving it somehow)

Maybe topic for a pair-programming session tomorrow?